### PR TITLE
Prow integration test: make it as required

### PR DESCRIPTION
**Pull request by chaodaiG**
Now that https://github.com/kubernetes/test-infra/pull/20708 is merged, re-enabling integration test as gating

### CODE REVIEW

**Comment by chaodaiG** [comment]
/hold
Until the test passed in this PR

**Comment by chaodaiG** [comment]
/unhold
Ready to merge
/cc @fejta

**Comment by fejta** [comment]
How long does this take:
a) On its own
b) When it also runs `bazel test //...` 

AKA Ideally I would like to merge this testing into `pull-test-infra-bazel` like all our other testing.

**Comment by chaodaiG** [comment]
> How long does this take:
> a) On its own

https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-test-infra-integration, ranging from 7 - 12 minutes, and I think a rough media is 8.5 minutes.

> b) When it also runs `bazel test //...`

Haven't tested, created a draft PR for measuring https://github.com/kubernetes/test-infra/pull/21044

**Comment by chaodaiG** [comment]
> > How long does this take:
> > a) On its own
> 
> https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-test-infra-integration, ranging from 7 - 12 minutes, and I think a rough median is 8.5 minutes.
> 
> > b) When it also runs `bazel test //...`
> 
> Haven't tested, created a draft PR for measuring #21044

**Comment by chaodaiG** [comment]
https://github.com/kubernetes/test-infra/pull/21044 finished https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365374671578468352, glad it worked. But the time it took was 12m20s which was almost double the current median of test-infra-bazel https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-test-infra-bazel

**Comment by fejta** [comment]
IMO the more relevant question is: how long does the integration job take when it isn't also running all the unit tests?

**Comment by chaodaiG** [comment]
When the integration test run by itself:
```
//prow/test/integration/test:go_default_test                             PASSED in 60.4s
```
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365095477774651392#1:build-log.txt%3A8996

When it also runs all other tests:
```
//prow/test/integration/test:go_default_test                             PASSED in 156.3s
```
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365374671578468352#1:build-log.txt%3A9417

**Comment by chaodaiG** [comment]
> IMO the more relevant question is: how long does the integration job take when it isn't also running all the unit tests?

Maybe I misunderstood your question, the integration job takes 12m20s when it also runs all the unit tests https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365374671578468352

**Comment by fejta** [comment]
> the integration job takes 12m20s when it also runs all the unit tests

Cool! And how long does it take when it doesn't run the test?


AKA the fact that pull-test-infra-bazel only takes 5m isn't particularly relevant if this job takes > 5m (since every job will wait however long this job takes)

**Comment by chaodaiG** [comment]
> Cool! And how long does it take when it doesn't run the test?
> 
> AKA the fact that pull-test-infra-bazel only takes 5m isn't particularly relevant if this job takes > 5m (since every job will wait however long this job takes)

good point, so integration test will actually make PR turnaround time longer, let me do more measurement here to make sure it's accurate

/test pull-test-infra-integration

**Comment by chaodaiG** [comment]
7m22s https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365395184610709504
/test pull-test-infra-integration

**Comment by chaodaiG** [comment]
7m46s https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365397679881850880
/test pull-test-infra-integration

**Comment by chaodaiG** [comment]
9m15s https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365400918589181952
/test pull-test-infra-integration

**Comment by chaodaiG** [comment]
9m2s https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365403753800273920
/test pull-test-infra-integration

**Comment by chaodaiG** [comment]
8m41s https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365407906685521920
/test pull-test-infra-integration

**Comment by chaodaiG** [comment]
12m9s https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365415190664843264
/test pull-test-infra-integration

**Review by fejta — APPROVED** [review]
/hold

Personally I would prefer all testing to happen in the single pull-test-infra-bazel job as this is IMO cognitively easier for everyone to understand and follow. Also puts lets IOPS pressure on the overall system.

Saving a minute or two IMO isn't worth it: I'd consider 1*12m is better than 2*10m jobs.

That being said this is also fine if you need/want to go the two job route.

**Comment by chaodaiG** [comment]
TLDR: I would prefer the two jobs route.

The reason was based off of the stats that I have collected by comparing running integration tests alone in current PR (this is going to be new bottleneck in test-infra), and running integration tests together with all unit tests as in https://github.com/kubernetes/test-infra/pull/21044. Performed 7 runs on each PR, and got:

Running integration tests alone:
[7m22s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365395184610709504) [7m46s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365397679881850880) [9m15s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365400918589181952) [9m2s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365403753800273920) [8m41s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365407906685521920) [12m9s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365415190664843264) [7m36s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21036/pull-test-infra-integration/1365419518251241472)

Running integration tests with unit tests:
[12m20s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365374671578468352) [18m54s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365383348578422784) [12m39s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365388607161896960) [12m1s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365397741399707648) [10m28s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365401019470581760) [10m24s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365405811035082752) [13m8s](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21044/pull-test-infra-integration/1365409724618510336)

The median time of running alone is ~8 minutes, and running together is ~12 minutes. If running these as two separate jobs doesn't make them more flaky (aka running into high iops), I would prefer faster turnaround time at the moment. 

That being said, if it turned out that this is a bad decision, switching to a single job isn't too hard https://github.com/kubernetes/test-infra/pull/21044
